cmd/rook/ceph/admission.go

```diff
@@ -36,11 +45,12 @@ func init() {
 
 func startAdmissionController(cmd *cobra.Command, args []string) {
 	rook.SetLogLevel()
-
 	rook.LogStartupInfo(admissionCmd.Flags())
 
-	context := rook.NewContext()
-
-	a := admission.New(context, "ceph", operator.ValidateCephResource)
+	err := cephv1.AddToScheme(scheme)
+	if err != nil {
+		logger.Errorf("failed to add to scheme %v. ", err)
+	}
+	a := admission.New("ceph", port, certDir, &cephv1.CephCluster{}, scheme)
```

Comment:
If we want to implement the admission webhook for multiple CRDs, we would need to start them all here (in separate goroutines, right? I guess they would just each listen on a separate port.

Reply:
Though i havent tested it yet but i think that may not be the right approach as adding new ports means adding more services + validatingwebhookconfigs+ adding each port on the deployment for each of the CRDs to expose the port.

Reply:
I have updated the code to add CephBlockPool validation. I've tried to make the code as generic and easy to change to add more CRDs. Here are the steps i did for cephblockpool which can be used for other ceph CRDs.

1.  Implement `webhook.Validator` interface in `pkg/apis/ceph.rook.io/v1/webhook.go` for the CRD which needs to be validated
2. Add a `ValidatingWebhookConfiguration` (copy-paste work) in `webhook-config.yaml` changing the relevant fields ( name, path, resources)  and make sure to note down the `path` given here.
3. In `cmd/ceph/admission.go', Add the above `path` and map it to the given CRD in `pathResourceMap`.
4. Done ! Test it out.

Reply:
Similiar approach can be followed for other providers also.